cli/tests/unit/command_test.ts

```diff
@@ -885,3 +888,60 @@ Deno.test(
     );
   },
 );
+
+// Test case for #19401
+Deno.test(
+  async function commandLogClosedStdout() {
+    const child = new Deno.Command(Deno.execPath(), {
+      args: ["run", "-A", "cli/tests/testdata/run/console_boom.js"],
+      stdout: "piped",
+    }).spawn();
+
+    // Create our own TextDecoderStream class to work around a spec issue
+    // where streams leak resources. See: https://github.com/whatwg/streams/issues/1212
+    class TextDecoderStream {
```

Comment:
This is a bit unfortunate, but there is a speak issue which leads to resource leaks. This is the most pragmatic workaround for now as it allows us to free the resource manually until the spec issue is addressed.